presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SimplifyCountOverConstant.java

```diff
@@ -0,0 +1,112 @@
+import com.google.common.collect.ImmutableList;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+
+import static com.facebook.presto.metadata.FunctionKind.AGGREGATE;
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+
+public class SimplifyCountOverConstant
+        implements Rule
+{
+    @Override
+    public Optional<PlanNode> apply(PlanNode node, Lookup lookup, PlanNodeIdAllocator idAllocator, SymbolAllocator symbolAllocator)
+    {
+        if (!(node instanceof AggregationNode)) {
+            return Optional.empty();
+        }
+
+        AggregationNode parent = (AggregationNode) node;
+
+        PlanNode input = lookup.resolve(parent.getSource());
+        if (!(input instanceof ProjectNode)) {
+            return Optional.empty();
+        }
+
+        ProjectNode child = (ProjectNode) input;
+
+        boolean changed = false;
+        Map<Symbol, FunctionCall> aggregations = new LinkedHashMap<>(parent.getAggregations());
+        Map<Symbol, Signature> functions = new LinkedHashMap<>(parent.getFunctions());
+
+        for (Entry<Symbol, FunctionCall> entry : parent.getAggregations().entrySet()) {
+            Symbol symbol = entry.getKey();
+            FunctionCall functionCall = entry.getValue();
+            Signature signature = parent.getFunctions().get(symbol);
+
+            if (isCountOverConstant(functionCall, signature, child.getAssignments())) {
+                changed = true;
+                aggregations.put(symbol, new FunctionCall(
+                        functionCall.getName(),
+                        functionCall.getWindow(),
+                        functionCall.getFilter(),
+                        functionCall.isDistinct(),
+                        ImmutableList.of()));
+
+                functions.put(symbol, new Signature("count", AGGREGATE, parseTypeSignature(StandardTypes.BIGINT)));
+            }
+        }
+
+        if (!changed) {
+            return Optional.empty();
+        }
+
+        return Optional.of(new AggregationNode(
+                node.getId(),
+                child,
+                aggregations,
+                functions,
+                parent.getMasks(),
+                parent.getGroupingSets(),
+                parent.getStep(),
+                parent.getHashSymbol(),
+                parent.getGroupIdSymbol()));
+    }
+
+    private static boolean isCountOverConstant(FunctionCall function, Signature signature, Assignments inputs)
+    {
+        if (!signature.getName().equals("count") || signature.getArgumentTypes().size() != 1) {
+            return false;
+        }
+
+        Expression argument = function.getArguments().get(0);
+        if (argument instanceof SymbolReference) {
+            argument = inputs.get(Symbol.from(argument));
+        }
+
+        return argument instanceof Literal && !(argument instanceof NullLiteral);
```

Comment:
has `select count(a) from (select 1 as a)` have count on constant? To me yest, it is kind of information that planner could leverage. So checking if expression is just a `Literal` is not enough to me.

To solve that I see two options:
 - introduce some kind of traits mechanism for plan nodes and store there information that given symbol is constant and store its values there as well
 - or replace constant symbol references in expressions with a literal